x-pack/plugins/enterprise_search/server/utils/create_pipeline_definitions.test.ts

```diff
@@ -34,3 +35,163 @@ describe('createIndexPipelineDefinitions util function', () => {
     expect(mockClient.ingest.putPipeline).toHaveBeenCalledTimes(3);
   });
 });
+
+describe('formatMlPipelineBody util function', () => {
+  const modelId = 'my-model-id';
+  let modelInputField = 'my-model-input-field';
+  const modelType = 'my-model-type';
+  const modelVersion = 3;
+  const sourceField = 'my-source-field';
+  const destField = 'my-dest-field';
+
+  const mockClient = {
+    ml: {
+      getTrainedModels: jest.fn(),
+    },
+  };
+
+  beforeEach(() => {
+    jest.clearAllMocks();
+  });
+
+  it('should return the pipeline body', async () => {
+    const expectedResult = {
+      description: '',
+      version: 1,
+      processors: [
+        {
+          remove: {
+            field: `ml.inference.${destField}`,
+            ignore_missing: true,
+          },
+        },
+        {
+          inference: {
+            model_id: `${modelId}`,
+            target_field: `ml.inference.${destField}`,
+            field_map: {
+              sourceField: modelInputField,
+            },
+          },
+        },
+        {
+          append: {
+            field: '_source._ingest.processors',
+            value: [
+              {
+                type: modelType,
+                model_id: modelId,
+                model_version: modelVersion,
+                processed_timestamp: '{{{ _ingest.timestamp }}}',
+              },
+            ],
+          },
+        },
+      ],
+    };
+
+    const mockResponse = {
+      count: 1,
+      trained_model_configs: [
+        {
+          model_id: modelId,
+          version: modelVersion,
+          model_type: modelType,
+          input: { field_names: [modelInputField] },
+        },
+      ],
+    };
+    mockClient.ml.getTrainedModels.mockImplementation(() => Promise.resolve(mockResponse));
+    const actualResult = await formatMlPipelineBody(
+      modelId,
+      sourceField,
+      destField,
+      mockClient as unknown as ElasticsearchClient
```

Comment:
Why do we use a double `as`?

Reply:
Otherwise you can't force this mock into the type. It doesn't mock enough. It mocks enough for our purposes.

Reply:
> It doesn't mock enough.

😂